src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/PremisesController.kt

```diff
@@ -175,6 +182,45 @@ class PremisesController(
     return ResponseEntity.ok(arrivalTransformer.transformJpaToApi(arrival))
   }
 
+  override fun premisesPremisesIdBookingsBookingIdNonArrivalsPost(
+    premisesId: UUID,
+    bookingId: UUID,
+    body: NewNonarrival
+  ): ResponseEntity<Nonarrival> {
+    val premises = premisesService.getPremises(premisesId)
+      ?: throw NotFoundProblem(premisesId, "Premises")
+
+    val booking = bookingService.getBooking(bookingId)
+      ?: throw NotFoundProblem(bookingId, "Booking")
+
+    if (booking.premises.id != premises.id) {
+      throw NotFoundProblem(bookingId, "Booking")
+    }
```

Comment:
Again, as with #54, it'd be good to reuse this stuff

Reply:
Will tackle this alongside the existing endpoints in a follow up PR